### DISCUSSION TITLE
Production environment variables in Preview deployments with create-react-app

### DISCUSSION THREAD

**Discussion by kcbarry** [vercel/vercel#5389] [Help] [answered]
When pushing to a test branch, I'd like to configure my frontend code to point at a different database url for product, test and dev.

I've seen this done by querying `process.env.NODE_ENV` but when using Vercel Preview, I have not been able to successfully query the test database because preview deployments share the same build command as the product deployments.

From my reading, it seems like the recommended way to do this would be to use the Environment Variables UI in my Project settings. This works as excepted for my serverless functions but I have not been able to figure out how to inject the Preview Environment variables into my frontend code during the build step. 

Any guidance on the correct approach here would be greatly appreciated, thanks!

**Comment by mcsdevv** [comment, marked as answer]
Hello @kcbarry, when using CRA, you should prefix environment variable names with `REACT_APP: https://create-react-app.dev/docs/adding-custom-environment-variables/

**Comment by reyesGeorge** [comment]
You may also need to install webpack.
